src/basic/cgroup-util.c

```diff
@@ -624,11 +624,7 @@ int cg_get_path(const char *controller, const char *path, const char *suffix, ch
         if (!cg_controller_is_valid(controller))
                 return -EINVAL;
 
-        unified = cg_all_unified();
-        if (unified < 0)
-                return unified;
-
-        if (unified > 0)
+        if (cg_all_unified())
```

Comment:
I don't like this change. You do get a nice reduction in LOC, but I don't think it is enough to justify it. We should not assert on external conditions, especially the ones like these, which depend on a lot of details in the environment and might legitimately be unmet. In particular, selinux can interfere with path access. `assert` is also a crappy user interface (you get a coredump, which is useless, because it's the external conditions that are wrong, not the internal program state), and it should not be used for checking arguments or external state or anything the user has control over.